src/biguint/iter.rs

```diff
@@ -181,9 +181,9 @@ pub struct U64Digits<'a> {
 }
 
 #[cfg(not(u64_digit))]
-impl<'a> U64Digits<'a> {
+impl U64Digits<'_> {
     #[inline]
-    pub(super) fn new(data: &'a [u32]) -> Self {
+    pub(super) fn new(data: &[u32]) -> Self {
```

Comment:
This also fails on 32-bit, but it passes with `-> U64Digits<'_>` to have a different anonymous lifetime than the `impl`'s `Self`. But that doesn't work with the other case, for some reason...